benchmark/script.py

```diff
@@ -56,33 +31,48 @@ class BenchmarkOptions(Options):
          'IP address for a Flocker cluster control server.'],
         ['certs', None, 'certs',
          'Directory containing client certificates'],
-        ['scenario', None, _DEFAULT_SCENARIO,
-         'Environmental scenario under which to perform test. '
-         'Supported values: {}.'.format(', '.join(_SCENARIOS))],
-        ['operation', None, _DEFAULT_OPERATION,
-         'Operation to measure. '
-         'Supported values: {}.'.format(', '.join(_OPERATIONS))],
-        ['metric', None, _DEFAULT_METRIC,
-         'Quantity to benchmark. '
-         'Supported values: {}.'.format(', '.join(_METRICS))],
+        ['config', None, 'benchmark.yml',
+         'YAML file describing benchmark options.'],
+        ['scenario', None, 'default',
+         'Environmental scenario under which to perform test.'],
+        ['operation', None, 'default', 'Operation to measure.'],
+        ['metric', None, 'default', 'Quantity to benchmark.'],
     ]
 
 
-config = BenchmarkOptions()
+options = BenchmarkOptions()
+
+
+def usage(message=None):
+    sys.stderr.write(options.getUsage())
+    sys.stderr.write('\n')
+    sys.exit(message)
+
 try:
-    config.parseOptions()
-    if not config['control'] and config['operation'] != 'no-op':
-        # No-op is OK with no control service
-        raise UsageError('Control service required')
+    options.parseOptions()
 except UsageError as e:
-    sys.stderr.write(config.getUsage())
-    sys.stderr.write('\n{}\n'.format(str(e)))
-    sys.exit(1)
+    usage(e.args[0])
+
+if not options['control'] and options['operation'] != 'no-op':
+    # No-op is OK with no control service
+    usage('Control service required')
+
+with open(options['config'], 'rt') as f:
+    config = yaml.safe_load(f)
+    scenarios = config['scenarios']
+    operations = config['operations']
+    metrics = config['metrics']
```

Comment:
It would be good to add validation of this configuration file. If a particular section in the file has an invalid configuration, it will result in a `KeyError` when it is used. When this error is encountered, it's unclear what the cause is. For example, if the key `type` is missing for any of the `scenarios`, `operations`, or `metrics` then the error will not be encountered until the `driver` has started.